### ISSUE TITLE
RS0041 code fix doesn't fix all issues, then doesn't show for the remainder

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn-analyzers#4028]
### Describe the bug

Upgrading Microsoft.CodeAnalysis.PublicApiAnalyzers from 3.3.0-beta2.final to 3.3.0 in vs-threading produces many more warnings. When I use the IDE code fix to bulk fix all in the solution, only a subset are fixed. The remainder still produce warnings and no code fix is offered to address them. 

See https://github.com/microsoft/vs-threading/pull/660

### Steps To Reproduce

```
git clone https://github.com/microsoft/vs-threading.git
cd vs-threading
git checkout dependabot/nuget/src/Roslyn.Diagnostics.Analyzers-3.3.0
src\Microsoft.VisualStudio.Threading.sln
```

Observe error list, navigate to one of the RS0041 warnings and apply a bulk fix on the solution.
Notice the error list is still not empty, but the RS0041 code fix provider doesn't show up any more.

### Expected behavior

The RS0041 code fix is offered for every issue, and fixes all issues.

### Actual behavior

Only some issues are fixed, leaving me to wonder how to get the rest fixed.

### Analyzer

**Package**: [Microsoft.CodeAnalysis.PublicApiAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.PublicApiAnalyzers)

**Version**: v3.3.0 (Latest)

**Diagnostic ID**: [RS0041](https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md)

**Comment by jcouv** [comment]
@AArnott What builk fix do you mean?
RS0041 is "Public members should not use oblivious types". There is no FixAll for that. It must either be addressed (ie. code annotated) or suppressed.

**Comment by AArnott** [comment]
@jcouv  then we might be talking about a bunch of false positives here. It's calling out enums, among other things. I have `#nullable enable` on all my code.

**Comment by Evangelink** [comment]
@AArnott I suspect this relates to the issue we faced internally on this repo, see https://github.com/dotnet/roslyn-analyzers/issues/3915. @jcouv if that's confirmed, there might be a need to improve the analyzer to avoid the same "bad" behavior I suspect a lot of devs might see.

**Comment by jcouv** [comment]
@AArnott Yes, there was a bug with enums ([nested enums](https://github.com/dotnet/roslyn-analyzers/pull/3425), specifically) which was fixed.
I tried the repro steps you provided, but couldn't load the solution (projects fail to load). Here's a sample of the errors, in case you have a tip for addressing the issue:
```
D:\repos\vs-threading\src\Microsoft.VisualStudio.Threading.Analyzers\Microsoft.VisualStudio.Threading.Analyzers.csproj : error  : The project file cannot be opened. Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.

D:\repos\vs-threading\src\Microsoft.VisualStudio.Threading.Analyzers.Tests\Microsoft.VisualStudio.Threading.Analyzers.Tests.csproj : error  : The project file cannot be opened. Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.

D:\repos\vs-threading\src\Microsoft.VisualStudio.Threading.Tests.Win7RegistryWatcher\Microsoft.VisualStudio.Threading.Tests.Win7RegistryWatcher.csproj : error  : The project file cannot be opened. Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.

D:\repos\vs-threading\src\Microsoft.VisualStudio.Threading\Microsoft.VisualStudio.Threading.csproj : error  : The project file cannot be opened. Unable to locate the .NET Core SDK. Check that it is installed and that the version specified in global.json (if any) matches the installed version.
```

Here's what the .NET SDKs I have installed:
```
D:\repos\vs-threading>dotnet --info

Host (useful for support):
  Version: 5.0.0-preview.7.20364.11
  Commit:  53976d38b1

.NET SDKs installed:
  3.1.400-preview-015203 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.7.20366.6 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-preview.7.20365.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-preview.7.20364.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-preview.7.20366.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by AArnott** [comment]
> but couldn't load the solution (projects fail to load).

Sorry. I left a step out of the repro steps (running init, which installs the necessary .NET Core SDK). I've updated the repro steps.

**Comment by AArnott** [comment]
Any update on this? I'm not sure how to finish upgrading from the prerelease analyzers to the stable analyzer with this bug active.

**Comment by AArnott** [comment]
I have another false positive: exactly one RS0041 and one RS0036 show up after upgrading `Microsoft.CodeAnalysis.PublicApiAnalyzers` to 3.3.0 (from 3.3.0-beta1.final) in [this repo](https://dev.azure.com/devdiv/DevDiv/_git/VS.RPC.Contracts). `Nullable=enable` is set for the entire repo, and yet it claims I'm not annotating this declaration:

```cs
    public class DiscriminatedTypeJsonConverter<TBase> : JsonConverter<TBase?>
        where TBase : class
```

Only RS0036 offers a code fix, but it 'fixes' it by adding `~` in front of that declaration in PublicAPI.Shipped.txt, and then RS0041 remains.

**Comment by jcouv** [comment]
Likely fixed by https://github.com/dotnet/roslyn-analyzers/pull/4246. Will try to verify

**Comment by AArnott** [comment]
@jcouv I just tried 3.3.1 and the bug remains. I see dozens of RS0041 errors for oblivious types in code that compiles with `Nullable=true` in the repo given in the repro steps.

**Comment by AArnott** [comment]
I just tried 3.3. and the bug remains. This is a blocking bug as there seems to be no way to satisfy the analyzer. Please fix.

**Comment by AArnott** [comment]
I just tried 3.3.3 and this still repros. :(
